### ISSUE TITLE
Is there any improvement on Sony cameras?

### ISSUE THREAD

**Issue by minw428** [gphoto/gphoto2#192]
I read all thread related to Sony.
I'm facing a same problem that the capture command is suspended after taking some shots.

As I know there are a few way to exit that state.
1. kill the process. You can see using 'ps aux' or Ctrl - c
2. take a shot manually using button of the camera  --> this is not working..  :(

Is there any improvement on this?
I'm using RX100M4 and it is hard to complete a timelapse.

**Comment by msmeissn** [comment]
"some shots" is how many usually?

this should not happen though, but I did not work on this as I was not fully aware of it.

Is it possible to capture a debug trace? 

Use --debug-level=debug   to not include data in the logfile

**Comment by minw428** [comment]
Sometimes over 100 shots, sometimes under 30 shots.

I think that there are some missing(or ignored by camera) command. Because when it happens if I press a shot button of camera then the suspended status is cleared.

**Comment by msmeissn** [comment]
yeah, likely a missed event or similar :/

**Comment by minw428** [comment]
Same result... :(
Can you recommend a flawless camera model?

**Comment by msmeissn** [comment]
can you try this patch... it enabled the event handling again, which should report on added images

reason it was disabled is that it was unstable on some models though

[patch.txt](https://github.com/gphoto/gphoto2/files/2735481/patch.txt)

**Comment by msmeissn** [comment]
The Nikon DSLR/Z and Canon EOS DSLR / R are more reliable FWIW

**Comment by minw428** [comment]
> can you try this patch... it enabled the event handling again, which should report on added images
> 
> reason it was disabled is that it was unstable on some models though
> 
> [patch.txt](https://github.com/gphoto/gphoto2/files/2735481/patch.txt)

compile error due to the "dual" variable.
if (dual)
   ptp_add_event (params, &event);

what "dual" should be?

**Comment by msmeissn** [comment]
it is ptp_check_event_queue in the patch, did oyu mistype it?

**Comment by minw428** [comment]
> it is ptp_check_event_queue in the patch, did oyu mistype it?

sorry :)

What should I do for "dual"?
https://github.com/gphoto/libgphoto2/blob/412ef5ef34afb59a921d1033825867e72b0e1b21/camlibs/ptp2/library.c#L4197

**Comment by maxbcn** [comment]
Hello Marcus, I had two cameras a5000 doing time-lapse and discover that the setting on the camera sould be on " PC REMOTE " not in ptp. With this setting it did work for me for a few days in one of the cameras, but now both cameras not doing nothing. I can send you one of my cameras for you to test if you need it and resolve this issue more deeply. 
Doing " lsusb " it appears the sony camera but is not working at all. Can you help us?

Marcos

**Comment by msmeissn** [comment]
if it survives several days of shooting this will be very hard to debug. :/

jhas it totally stopped working?

**Comment by maxbcn** [comment]
yes , I have it by ssh to a static IP , do you want to tes it, is in position PC REMOTE

**Comment by maxbcn** [comment]
Yes , stopped with not reason aparenly

**Comment by maxbcn** [comment]
I restart the camera and did only one picture and stop all.
Attached log file debug
[my-logfile.txt](https://github.com/gphoto/gphoto2/files/2849434/my-logfile.txt)

On "lsusb" I can see the camera.

Thank you

**Comment by Brechtjeu** [comment]
Hi, 

I have something, that is not really a permanent or pretty solution but rather a hot fix.
I opened issue #216, about the same problem, where I'm using python. At this time I can recover from this by executing  

> kill -9 `ps aux | grep gphoto | grep -v grep | awk '{print $2}'`

each time the capture process takes longer than 6-7 seconds to terminate. (using multiprocessing in python)

it ain't pretty but it works...
